### ISSUE TITLE
Allow line breaks in fields

### ISSUE THREAD

**Issue by ukutaht** [beatrichartz/csv#13]
According to [RFC 4180](http://tools.ietf.org/html/rfc4180) page 2, fields that are enclosed in double quotes can have newlines in them. When trying to parse such row:
```
first,row,here
one,two,"three
and newline"
```

The parser throws a syntax error:
```
(CSV.Parser.SyntaxError) Unterminated escape sequence. on line 1
```

Notice that the reported line number is incorrect as well – confusing.

I'll take a look if I can fix it, but I may need some direction :)

**Comment by beatrichartz** [comment]
Sorry for the delayed update - I thought about this and it turns out a fix would be rather intrusive. This does not mean that this is a wontfix, but I would like to make sure that the implementation (and maybe the additional API) is well thought out.

The problem is that atm, the input to CSV is a stream of lines of a file - something that almost definitely would have to change. 

An implementation with line-based input would require some kind of memorization of previous lines starting from the line with the starting quote, which is not something I want to implement. So the solution is to require a byte-stream as input and to transfer line separation to the parser.

Another issue is that indeterminate escape sequences, could lead to whole files being sunk into memory - after all, there is no limit to how big a field can be. This can be fixed by providing an (arbitrary) upper limit to the field byte-size and make that configurable.

I'll move these to the `2.0.0` milestone - I hope I can provide a fix in the next months - Thanks again for reporting!

**Comment by seamusabshere** [comment]
> An implementation with line-based input would require some kind of memorization of previous lines starting from the line with the starting quote, which is not something I want to implement. 

why in particular do you not want to do this?

**Comment by beatrichartz** [comment]
@seamusabshere Loosing the 1:1 mapping of stream output to rows would complicate things throughout the library.

@ukutaht @seamusabshere I added multiline support in master, however I'd like to test it a bit more before releasing. It should handle multiline escapes pretty decently already, let me know if you find something.

**Comment by beatrichartz** [comment]
@seamusabshere @ukutaht This is released in `1.3`
